lib/digest/crc.rb

```diff
@@ -93,10 +96,52 @@ def digest_length
     # @param [String] data
     #   The data to update the CRC checksum with.
     #
-    # @abstract
-    #
     def update(data)
-      raise(NotImplementedError,"#{self.class}##{__method__} not implemented")
+      unless @width >= 1
+        raise(NotImplementedError, "incompleted #{self.class} as CRC (expected WIDTH to be 1 or more)")
+      end
+
+      if @table.empty?
+        raise(NotImplementedError, "incompleted #{self.class} as CRC (expected TABLE to be not empty)")
+      end
+
+      if self.class::REFLECT_INPUT.nil?
+        raise(NotImplementedError, "incompleted #{self.class} as CRC (expected REFLECT_INPUT to be not nil)")
+      end
```

Comment:
This logic could use some specs. Just need to define a bunch of incorrect `CRC` classes in the specs:

```ruby
module TestCRC
  class CRCWithZeroWidth < Digest::CRC
    WIDTH = 0
  end
end
subject { TestCRC::CRCWithZeroWidth.new }
```

Reply:
Could I have made the changes you intended?
https://github.com/postmodern/digest-crc/pull/41/files#diff-525bf8ab75d7dddf72278222e8cbf5b2bf2f1437dbd1d368337c965d240f964bR82-R135